src/plugins/irc-events/error.js

```diff
@@ -40,4 +40,19 @@ module.exports = function(irc, network) {
 		var random = irc.user.nick + Math.floor(10 + (Math.random() * 89));
 		irc.changeNick(random);
 	});
+
+	irc.on("nick invalid", function(data) {
+		var lobby = network.channels[0];
+		var msg = new Msg({
+			type: Msg.Type.ERROR,
+			text: "Nickname " + data.nick + " is invalid: " + data.reason,
+		});
+		client.emit("msg", {
+			chan: lobby.id,
+			msg: msg
+		});
+
+		var random = "i" + Math.random().toString(36).substr(2, 10); // 'i' so it never begins with a number
```

Comment:
Testing on `v1.4.3` vs testing on `irc-framework` branch, if changing to an invalid nick:
- on `v1.4.3`, command is denied and error on lobby is `Erroneous Nickname`
- on `irc-framework`, my nick got changed to `i0qugh5endq`, very odd lol

I think `/nick` command should not be executed when nick is invalid, essentially what `v1.4.3` is currently doing. Not sure what happens / should happen when connecting to the network (as opposed to running `/nick`).